crates/circuit/src/symbol_expr.rs

```diff
@@ -0,0 +1,3214 @@
+                    Value::Int(t as i64)
+                } else {
+                    Value::Real(t)
+                }
+            }
+            Value::Complex(e) => {
+                let t = Value::Complex(e.sqrt());
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn pow(&self, p: &Value) -> Value {
+        match self {
+            Value::Real(e) => match p {
+                Value::Real(r) => {
+                    if *e < 0.0 {
+                        Value::Complex(Complex64::from(e).powf(*r))
+                    } else {
+                        Value::Real(e.powf(*r))
+                    }
+                }
+                Value::Int(i) => {
+                    if *e < 0.0 {
+                        Value::Complex(Complex64::from(e).powf(*i as f64))
+                    } else {
+                        Value::Real(e.powf(*i as f64))
+                    }
+                }
+                Value::Complex(r) => Value::Complex(Complex64::from(e).powc(*r)),
+            },
+            Value::Int(e) => {
+                if *e < 0 {
+                    match p {
+                        Value::Real(r) => Value::Complex(Complex64::from(*e as f64).powf(*r)),
+                        Value::Int(i) => Value::Complex(Complex64::from(*e as f64).powf(*i as f64)),
+                        Value::Complex(c) => Value::Complex(Complex64::from(*e as f64).powc(*c)),
+                    }
+                } else {
+                    match p {
+                        Value::Real(r) => {
+                            let t = (*e as f64).powf(*r);
+                            let d = t.floor() - t;
+                            if (-f64::EPSILON..f64::EPSILON).contains(&d) {
+                                Value::Int(t as i64)
+                            } else {
+                                Value::Real(t)
+                            }
+                        }
+                        Value::Int(r) => {
+                            if *r < 0 {
+                                Value::Real((*e as f64).powf(*r as f64))
+                            } else {
+                                Value::Int(e.pow(*r as u32))
+                            }
+                        }
+                        Value::Complex(r) => Value::Complex(Complex64::from(*e as f64).powc(*r)),
+                    }
+                }
+            }
+            Value::Complex(e) => {
+                let t = match p {
+                    Value::Real(r) => Value::Complex(e.powf(*r)),
+                    Value::Int(r) => Value::Complex(e.powf(*r as f64)),
+                    Value::Complex(r) => Value::Complex(e.powc(*r)),
+                };
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn rcp(&self) -> Value {
```

Comment:
I think it would be useful to add it to the python interface as well.

Reply:
This is only `1 / <expr>` though, right? I'm not sure we additionally need this on the Python interface, since there's already a way to do it, no?